boot/boot_serial/src/boot_serial.c

```diff
@@ -586,16 +582,21 @@ boot_serial_start(const struct boot_uart_funcs *f)
             }
             continue;
         }
-        if (buf[0] == SHELL_NLIP_PKT_START1 &&
-          buf[1] == SHELL_NLIP_PKT_START2) {
+        if (in_buf[0] == SHELL_NLIP_PKT_START1 &&
+          in_buf[1] == SHELL_NLIP_PKT_START2) {
             dec_off = 0;
-            rc = boot_serial_in_dec(&buf[2], off - 2, dec, &dec_off, max_input);
-        } else if (buf[0] == SHELL_NLIP_DATA_START1 &&
-          buf[1] == SHELL_NLIP_DATA_START2) {
-            rc = boot_serial_in_dec(&buf[2], off - 2, dec, &dec_off, max_input);
+            rc = boot_serial_in_dec(&in_buf[2], off - 2, dec_buf, &dec_off, max_input);
+        } else if (in_buf[0] == SHELL_NLIP_DATA_START1 &&
+          in_buf[1] == SHELL_NLIP_DATA_START2) {
+            rc = boot_serial_in_dec(&in_buf[2], off - 2, dec_buf, &dec_off, max_input);
+        }
+
+        /* serve errors: out of decode memory, or bad encoding */
+        if (rc == -1) {
```

Comment:
if this is not a diff issue, one empty block was left behind here (I see the correct handling just below)

Reply:
@nvlsianpu can you please reply since this is in one of your commits?

Reply:
Rightm, This if can be used for error in protocol parsing signaling, as it is empty - should be removed.

Reply:
@carlescufi It Is fixed on my fork.

Reply:
@utzig Fixed now